broadcast/util/sendmail.py

```diff
@@ -10,71 +10,79 @@
+from email.mime.multipart import MIMEMultipart
 
-try:
-    from urllib.parse import urlparse
-except ImportError:
-    from urlparse import urlparse
+from bottle import request, template
 
-import mandrill
-# mako_template must be used because simplates do not support multiline strings
-# and our custom template function has the request object in it's context,
-# which for deferred tasks such as email sending causes the:
-# 'This request is not connected to an application' error.
-from bottle import request, mako_template as template
 
+def send_multiple(to_list, subject, text=None, data={},
+                  is_async=False, config=None):
+    """
+    Sends out text/HTML email with specified templates
 
-def send_multiple(to_list, subject, text=None, html=None, data={},
-                  mandrill_args={'preserve_recipients': False},
-                  use_template=None, is_async=False, config=None):
-    """ Sends out text/HTML email with specified templates """
+    `to_list` is a list of tuples containing an email and a name. This
+    is historical from BD (Before Docstrings). `subject` is a single line
+    string used as the subject of the message. `text` is a string that
+    specifies what template to use in combination with the data. It is named
+    `text` and a keyword argument for historical reasons. `data` is a dict that
+    is passed to the template to create the message. `config` is an initialized
+    config object (see main module) if not provided, the call must be part of a
+    request context.
+    """
+    # As described in the docstring, you must provide a template
+    if text is None:
+        with Exception('no template specified') as ex:
+            logging.exception('Error sending email: %s' % ex)
+        return None
+
+    # Because this is a bottle app we can get the config through request
+    # context
     conf = config or request.app.config
 
-    # Set up mandrill API access
-    mandrill_client = mandrill.Mandrill(conf['mandrill.key'])
+    # Construct message object
+    msg = MIMEMultipart()  # Special object for building emails
+    msg['subject'] = subject
+    msg['from'] = conf['smtp.user']  # Get sender's email from the config
+    # As described in the docstring, we only use the first item in whatever
+    # objects make up the to_list. This is historical, BD (Before Docstrings).
+    msg['to'] = ', '.join([e[0] for e in to_list])
+
+    # Add the subject of the email and a newline character
+    msg.preamble = subject + '\n'
 
-    # Prepare data for the email
-    url = request.url if config is None else conf.get('app.url', '')
-    parsed = urlparse(url)
-    data['protocol'] = parsed.scheme
-    data['host'] = parsed.netloc
-    data['host_url'] = parsed.scheme + '://' + parsed.netloc
+    # Process the data with the chosen template
+    message = template(text, **data)
+    plain = MIMEText(message, 'plain', 'utf-8')
+    msg.attach(plain)
+    logging.debug("Prepared message")
```

Comment:
This log message would be better served if moved to `try ... catch` where the email is finally sent, with some of the parameters like the to field